### PR TITLE
feat: allow track component customization

### PR DESCRIPTION
**Pull request by pawelangelow**
It would be nice if there's a control over the Track look. I've played around and this is my solution to the problem. LMK if I'm missing a step or details (I couldn't find a contributing guide).

My example is not that smart 🤔, because the demo implementation can be achieved through `styles`. Another example would be a specific component, let's say a linear gradient - `<LinearGradient />` and the following usage 

```
const LinearGradient = () => <HeavyUiLogicComponent />

<Slider 
  renderMinimumTrackComponent={LinearGradient} 
  renderMaximumTrackComponent={LinearGradient}
/>
```

## Preview

https://user-images.githubusercontent.com/6314481/176449321-548bf74e-a27c-4ef4-9388-bdd05d9fdbdf.mov

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export type SliderProps = {
     onValueChange?: SliderOnChangeCallback;
     renderAboveThumbComponent?: (index: number) => React.ReactNode;
     renderThumbComponent?: () => React.ReactNode;
+    renderMinimumTrackMarkComponent?: () => React.ReactNode;
```
Comment (miblanchard) [inline src/types.ts]: This isn't really rendering a TrackMark vs. the minimum Track itself.  Could you rename this to renderMinimumTrackComponent and remove the `Mark`?

**Review by miblanchard — COMMENTED** [review]
This is cool.  Would you be able to add the same thing for renderMaximumTrackComponent please?  This could unlock some cool use cases.

File: src/types.ts (hunk 1 above)
Comment (pawelangelow) [reply src/types.ts]: Oh... 🤦 I guess I need more coffee 😸

**Comment by pawelangelow** [comment]
The preview for `renderMaximumTrackComponent`

https://user-images.githubusercontent.com/6314481/177112756-677f3692-282c-400a-9b5c-727cb3012698.mov

And a better demo for such a use case


https://user-images.githubusercontent.com/6314481/177112902-10f9f4b9-3698-4c8a-917e-a36f6a978546.mov

**Comment by miblanchard** [comment]
@pawelangelow - Very sorry for the delay here, would you be able to resolve the conflicts in this PR?

**Comment by pawelangelow** [comment]
@miblanchard Sure, no worries 👍 

Demo after rebase:


https://user-images.githubusercontent.com/6314481/209987105-e595a1df-63eb-473c-bb5d-5daf9811f1b7.mov

**Review by miblanchard — APPROVED** [review]

**Comment by 6ase** [comment]
Hi! Does it already available? Cannot understand how to use it. I am adding:

> renderMinimumTrackComponent={() =>
>                   <LinearGradient
>                   colors={[
>                               THEME_CONSTANTS.COLORS[theme].CIRCLE_SLIDER.START, 
>                               THEME_CONSTANTS.COLORS[theme].CIRCLE_SLIDER.END]} 
>                   start={[0, 0]}
>                   end={[1, 1]}
>                                 />}

But nouthing happend

**Pull request merged**